### PR TITLE
Add ID attribute for elements on the Feed Page

### LINKED ISSUE #1045: Feed Page: Add ID attribute for elements on this page
**Sorry, this is a MASSIVE one!!!**
**Please make sure the id attributes are updated for the mobile version as well!!**

Please assign the following elements with the listed JSX 'id' attributes.

Please also note, when assigning id attribute, to assign it to the area that encompasses the entire button or element (including any _clickable_ WHITE SPACE around text).
Example: not just the text "Next", but the entire button (including Next and arrow symbol)
Screen Shot 2020-07-03 at 6 43 37 PM

Why we need to assign unique IDs: [https://www.optimizesmart.com/event-tracking-in-google-tag-manager-v2-complete-guide/#:~:text=You%20will%20need%20this%20'id,not%20contain%20any%20white%20space.](url)

## Feed Page (All ID's prefix: HB_)

*All the id values relating to post component require a new id value to be generated per post. The id values should include the post's unique number saved in the database, as indicated by "_postnumber". Refer to #1031

| #  | Element on page | JSX 'id' attribute |
| --- | --- | --- |
| 1. | Like icon (including "X Likes" text) | HB_LK_postnumber |
| 2. | Comment icon + "Comment" text | HB_CM_postnumber |
| 3. | Share icon + "Share" text | HB_SH_postnumber |
| 4. | "View More" text | HB_VM_postnumber |
| 5. | "Create a Post" text and "+" icon | HB_CP |
| 6. | "All Posts" tab <br>(including clickable white space) | HB_AP |
| 7. | "Requesting Help" tab <br>(including clickable white space) | HB_RE |
| 8. | "Offering Help" tab <br>(including clickable white space) | HB_OF |
| 9. | Filter icon + "Filter" text | HB_FI |

**Desktop Ver:**
Screen Shot 2020-07-06 at 2 59 37 PM

**Mobile Ver:** (This one just shows the components that are different from desktop version)
<p float="left">
  
   
</p>

### Feed filters
| #  | Element on page | JSX 'id' attribute |
| --- | --- | --- |
| 10. | "Location" text and + Button <br>(including clickable white space) | HB_LO |
| 11. | "Providers" text and + Button <br>(including clickable white space) | HB_PV |
| 12. | "Type" text and + Button <br>(including clickable white space) | HB_TP |

Screen Shot 2020-07-06 at 2 59 37 PM

### Location filters
| #  | Element on page | JSX 'id' attribute |
| --- | --- | --- |
| 13. | "Enter address, zip code, or city" text field | HB_LO_EAZ |
| 14. | "Share my location" text and icon | HB_LO_SL |

Screen Shot 2020-07-06 at 2 59 44 PM
<br/>

### Providers filters
| #  | Element on page | JSX 'id' attribute |
| --- | --- | --- |
| 15. | "Individual" | HB_PV_IN |
| 16. | "Startup" | HB_PV_ST |
| 17. | "Company" | HB_PV_CMP |
| 18. | "Community" | HB_PV_CM |
| 19. | "Government" | HB_PV_GOV |
| 20. | "R&D" | HB_PV_RD |
| 21. | "Non-profit" | HB_PV_NP |
| 22. | "University" | HB_PV_UN |
| 23. | "Health care provider" | HB_PV_HCP |
| 24. | "Other" | HB_PV_OT |

Screen Shot 2020-07-06 at 2 59 52 PM
<br/>

### Type filters
| #  | Element on page | JSX 'id' attribute |
| --- | --- | --- |
| 25. | "Medical Supplies" | HB_TP_MS |
| 26. | "Groceries/Food" | HB_TP_GF |
| 27. | "Business" | HB_TP_BS |
| 28. | "Education" | HB_TP_EDU |
| 29. | "Legal" | HB_TP_LE |
| 30. | "Wellbeing/Mental" | HB_TP_WM |
| 31. | "Entertainment" | HB_TP_ENT |
| 32. | "Information" | HB_TP_IF |
| 33. | "Funding" | HB_TP_FU |
| 34. | "R&D" | HB_TP_RD |
| 35. | "Tech" | HB_TP_TC |
| 36. | "Other" | HB_TP_OT |

Screen Shot 2020-07-06 at 2 59 59 PM
<br/>

### Creating a post from the feed page
| #  | Element on page | JSX 'id' attribute |
| --- | --- | --- |
| 37. | Continue posting as Modal <br>(the entire div component) | HB_CP1 |
| 38. | Individual Button | HB_CP1_IN |
| 39. | Organisation Button | HB_CP1_ORG |

Screen Shot 2020-07-06 at 3 00 08 PM
<br/>

| #  | Element on page | JSX 'id' attribute |
| --- | --- | --- |
| 40. | "Create a new one" Button | HB_CP1_ORG_CN |
| 41. | Back icon | HB_CP1_ORG_BA |

Screen Shot 2020-07-06 at 3 00 16 PM
<br/>

| #  | Element on page | JSX 'id' attribute |
| --- | --- | --- |
| 42. | Offer Help/Request Help Modal <br>(the entire div component)| HB_CP2 |
| 43. | Offering Help Tab Button | HB_CP2_OF |
| 44. | Post Button | HB_CP2_OF_PO |

Screen Shot 2020-07-06 at 3 00 25 PM
<br/>

| #  | Element on page | JSX 'id' attribute |
| --- | --- | --- |
| 45. | Requesting Help Tab Button | HB_CP2_RE |
| 46. | Post Button | HB_CP2_RE_PO |

Screen Shot 2020-07-06 at 3 00 32 PM

### PR DESCRIPTION
**Pull request by Gtstride**
### What does this pull request aim to achieve? 💯
resolves #1045

### CODE REVIEW

File: client/src/components/Feed/PostSocial.js [hunk 1]
```diff
@@ -147,16 +147,17 @@ const PostSocial = ({
       <span></span>
 
       {postId ? (
-        <div id={gtmTag("share")} className="social-icon">
+        <div id={gtmTag("share", GTM.post.prefix)} className="social-icon">
           <CopyToClipboard text={url} onCopy={onCopyLink}>
             <span>
               {renderShareIcon()}
-              <span className="social-text">Share</span>
+              <span className="total-number">{numShares}</span>
```
Comment (jyclam) [inline client/src/components/Feed/PostSocial.js:154]: Did we decide to display the number of shares? (Also, don't know if it is actually working, please resolve merge conflicts first.

Comment (idjevm) [reply client/src/components/Feed/PostSocial.js:154]: Yeah this one is not ready @jyclam, can we work on this one now? I will resolve, remove extra commit and push in a few mins but afterwards we should work on the other stuff that is missing

Comment (idjevm) [reply client/src/components/Feed/PostSocial.js:154]: GT has head out for the night, and I will head out soon :(

Comment (omarzoghayyer) [reply client/src/components/Feed/PostSocial.js:154]: This branch doesn't seem to be complete and it has merge conflicts.

Comment (jyclam) [reply client/src/components/Feed/PostSocial.js:154]: @idjevm Sure, I can take over but I likely won't finish it until tomorrow.

**Comment by adisen** [comment]
Link to test?

**Comment by galeza** [comment]
https://featurefeed-page-id-attribute.fightpandemics.xyz

**Pull request merged**